File: Courzelo_Back/src/main/java/tn/esprit/courzelo/entities/ForumEntities/RateQuestion.java
package tn.esprit.courzelo.entities.ForumEntities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import tn.esprit.courzelo.entities.UserCorzelo.UserCourzelo;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "RateQuestion")

public class RateQuestion {
    @Id
    private String id;
    @Indexed
    private int rateNumber ;
    @Indexed
    @DBRef
    private QuestionForum questionForum;
    @Indexed
    @DBRef
    private UserCourzelo user;
}

File: Courzelo_Front/Courzelo/src/app/Models/ForumEntities/RateQuestion.ts
import { UserCourzelo } from "../UserCorzelo/UserCourzelo";
import { QuestionForum } from "./QuestionForum";


export class RateQuestion{
    id!:String;
    rateNumber!:number;
 
    questionForum!: Array<QuestionForum>;
    user!: Array<UserCourzelo>;
}